### PR TITLE
Add Hubert for audio classification

### PR DESCRIPTION
**Pull request by jimypbr**
* Compiling and running. Waiting on next poplar SDK to land to main.
* [x] Superb fine-tuning
* [x] Common-lang fine-tuning
* [x] Need to add recompilation checkpoints
* [x] Need to optimize the pipeline layout
* [ ] Layer dropout not supported
* [ ] SpecAugment not supported

### CODE REVIEW

**Comment by jimypbr** [comment]
- [x] Fine-tuning with `superb-ks` dataset
- [ ]  Fine-tuning with `commonlang` dataset

File: examples/audio-classification/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+very little annotated data to yield good performance on speech classification datasets.
+
+## SUPERB Dataset
+
+The following command shows how to fine-tune [hubert-base-ls960](https://huggingface.co/facebook/hubert-base-ls960) on the 🗣️ [Keyword Spotting subset](https://huggingface.co/datasets/superb#ks) of the SUPERB dataset.
+
+```bash
+python examples/audio-classification/run_audio_classification.py \
+    --model_name_or_path facebook/hubert-base-ls960 \
+    --ipu_config_name examples/audio-classification/hubert-base \
```
Comment (michaelbenayoun) [inline examples/audio-classification/README.md:34]: Put the hubert config hosted on the hub instead?

File: examples/audio-classification/README.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+### Examples
+
+The following table shows a couple of demonstration fine-tuning runs.
+It has been verified that the script works for the following datasets:
+
+- [SUPERB Keyword Spotting](https://huggingface.co/datasets/superb#ks)
+- [Common Language](https://huggingface.co/datasets/common_language)
+
+| Dataset | Pretrained Model | # transformer layers | Accuracy on eval | GPU setup | Training time | Fine-tuned Model & Logs |
```
Comment (michaelbenayoun) [inline examples/audio-classification/README.md:134]: Remove this table since it relates to GPUs?

File: tests/examples/run_audio_classification.txt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+>             audio["array"], max_length=data_args.max_length_seconds, sample_rate=feature_extractor.sampling_rate
+>         )
+>         max_length = int(round(feature_extractor.sampling_rate * data_args.max_length_seconds))
+>         inputs = feature_extractor(
+>             subsampled_audio,
+>             max_length=max_length,
+>             sampling_rate=feature_extractor.sampling_rate,
+>             padding="max_length",
+>         )
+>         # examples.update(inputs)
```
Comment (michaelbenayoun) [inline tests/examples/run_audio_classification.txt:71]: Maybe we should remove this comment from the example and from the diff?

**Review by michaelbenayoun — APPROVED** [review]
